orderbook/openapi.yml

```diff
@@ -265,6 +291,29 @@ components:
       allOf:
         - $ref: "#/components/schemas/OrderCreation"
         - $ref: "#/components/schemas/OrderMetaData"
+    Trade:
+      description: |
+        Trade data such as executed amounts, fees, order id and block number.
+      type: object
+      properties:
```

Comment:
Could we return also the hash of the
- tx (the settlement tx hash) --> people will be able to use it to check in etherscan
- Can we return batch Id: this batch id should be enough to identify a batch, and allow us to query info about it in future endpoints
- executionTime  ---> at what time this trade was executed
-

Reply:
Good point. We don't have a notion of batch ID yet, but it makes sense. I was thinking batch_id will likely be something like tx_hash + tx_index of the Settlement transaction

Reply:
As discussed in the meeting yesterday we can compute a TradeUID base on tx hash + log index of the trade event which we should also return here. Can you add these values?

Reply:
I included `batchId` as `blockNumber * 10^4 + log_index` because the maximum number of transactions that could ever fit into a single block is about 594. This is computed as `blockGasLimit / minimumGasForTx`

```
12500000/21000 ~ 595
```

I hope this is alright for everyone. @fleupold

Reply:
This also allows room for the block gas limit to go up to 21M

Reply:
I don't think it has to be packed this way because the type doesn't have to be integer. If you use hex encoded bytes it could be block number concatenated with log index. But then as Felix said we probably want transaction hash instead of block number to make it easier to look up the tx later. I'm also fine with figuring out and discussing this in another issue/PR .

Reply:
Yes, we are certainly going to need transaction hash since log_index is not what I thought it was. This will be better for the explorer since they can provide a link to etherscan for the tx

Reply:
I think this should be left out for now and included in a later feature branch. There is already enough going on here and it would be an easy fix afterwards.

Reply:
For some reason I was confusing it with transaction index which would have given us a nice Integral `batchId`. Maybe we should call it `batchTx` when we implement it.

Reply:
Note that I have included this as a TODO (after the fact) in the Trade API issue #308

Reply:
Ok to do it later, and in general we don't have to inform in the API docs what the semantics of a unique ID is (we do it for orderUid because it can be used in the contracts directly). As far as the user of this API is concerned it's some opaque string they can use to uniquely identify a trade,